### ISSUE TITLE
Updating Cargo.toml file doesn't change the lint

### ISSUE THREAD

**Issue by botahamec** [rust-lang/rust-clippy#8512]
### Summary

The `cargo_common_metadata` lint should update when the `Cargo.toml` file is changed. I wasn't able to get clippy to recognize the fact that my `Cargo.toml` file had the categories value in it until I saved my `lib.rs` file

### Lint Name

cargo_common_metadata

### Reproducer

I tried this code:

```toml
[package]
name = "botic"
description = "A time crate that handles leap seconds"
version = "0.1.0"
edition = "2021"
license = "Unlicense"
repository = "https://github.com/botahamec/botic"
keywords = ["date", "time", "calendar"]
categories = ["date-and-time", "parser-implementations", "value-formatting"]
```

I saw this happen:

```
warning: package `botic` is missing `package.categories` metadata
  |
note: the lint level is defined here
 --> src\lib.rs:2:9
  |
2 | #![warn(clippy::cargo)]
  |         ^^^^^^^^^^^^^
  = note: `#[warn(clippy::cargo_common_metadata)]` implied by `#[warn(clippy::cargo)]`
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#cargo_common_metadata

warning: `botic` (lib) generated 1 warning
```

I expected no warning to appear

### Version

```text
rustc 1.59.0 (9d1b2106e 2022-02-23)
binary: rustc
commit-hash: 9d1b2106e23b1abd32fce1f17267604a5102f57a
commit-date: 2022-02-23
host: x86_64-pc-windows-msvc
release: 1.59.0
LLVM version: 13.0.0
```

### Additional Labels

@rustbot -label I-false-negative
